### ISSUE TITLE
Custom navigation puck?

### ISSUE THREAD

**Issue by spotelf13** [mapbox/mapbox-navigation-ios#3084]
Hi there, how would one go about adding a custom user puck to replace the default arrow navigation puck?

**Comment by ShanMa1991** [comment]
Hi @spotelf13 , yes, you could update the user puck through the `NavigationMapView.userLocationStyle`. It has three types to choose, the default arrow navigation puck `userCourseView`, the `puck2D` and `puck3D`. You may need to provide your own configuration such as following:

```
       var configuration = Puck2DConfiguration()
        if #available(iOS 13.0, *) {
            configuration.topImage = UIImage(systemName: "arrow.up")
            configuration.scale = .constant(2.0)
        }
       navigationMapView.userLocationStyle = .puck2D(configuration: configuration)
```

or 
```
       let uri = Bundle.main.url(forResource: "someModelFile", withExtension: "gltf")
        let myModel = Model(uri: uri,
                            position: [-122.396152, 37.79129], // Coordinates of the model in `[longitude, latitude]`format
                            orientation: [0, 0, 90])
        let scalingExpression = someExpression // Setting an expression to  scale the model based on camera zoom
        let configuration = Puck3DConfiguration(model: myModel, modelScale: .expression(scalingExpression))
        navigationMapView.userLocationStyle = .puck3D(configuration: configuration)
```

**Comment by 1ec5** [comment]
Similarly, if you need to use v1._x_ before v2.0 is available for production usage, then you can set `NavigationMapView.userCourseView` to an instance of a class you define that inherits from UIView and conforms to UserCourseView.
